backend/httpclient/max_bytes_reader.go

```diff
@@ -24,13 +24,14 @@ var ErrResponseBodyTooLarge = errors.New("http: response body too large")
 // MaxBytesReader prevents clients from accidentally or maliciously
 // sending a large request and wasting server resources.
 func MaxBytesReader(r io.ReadCloser, n int64) io.ReadCloser {
-	return &maxBytesReader{r: r, n: n}
+	return &maxBytesReader{r: r, remainingBytes: n, actualLimit: n}
 }
 
 type maxBytesReader struct {
-	r   io.ReadCloser // underlying reader
-	n   int64         // max bytes remaining
-	err error         // sticky error
+	r              io.ReadCloser // underlying reader
+	remainingBytes int64         // max bytes remaining
+	actualLimit    int64         // the actual limit
```

Comment:
nit: `actualLimit` sounds to me like a limit that changes in time (better to just call it `limit`).